### PR TITLE
Add events for Cognito custom authentication challenge

### PR DESCRIPTION
**Pull request by mxpv**
This PR adds events for Cognito's custom authentication - https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-challenge.html

### CODE REVIEW

File: events/cognito.go [hunk 1]
```diff
@@ -124,3 +124,72 @@ type GroupConfiguration struct {
...
+	CognitoEventUserPoolsHeader
+	Request  CognitoEventUserPoolsCreateAuthChallengeRequest  `json:"request"`
+	Response CognitoEventUserPoolsCreateAuthChallengeResponse `json:"response"`
+}
+
+// CognitoEventUserPoolsVerifyAuthChallengeRequest defines verify auth challenge request parameters
+type CognitoEventUserPoolsVerifyAuthChallengeRequest struct {
+	UserAttributes             map[string]string `json:"userAttributes"`
+	PrivateChallengeParameters map[string]string `json:"privateChallengeParameters"`
+	ChallengeAnswer            string            `json:"challengeAnswer"`
```
Comment (bmoffatt) [inline events/cognito.go]: from https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-verify-auth-challenge-response.html#cognito-user-pools-lambda-trigger-syntax-verify-auth-challenge it looks like this needs to be either an `map[string]string` or an `interface{}`

**Comment by hgl** [comment]
Any chance this getting merge any time soon?

File: events/cognito.go (hunk 1 above)
Comment (mxpv) [reply events/cognito.go]: Fixed

**Comment by mxpv** [comment]
@bmoffatt I'm not quite sure whether `map[string]string` is the right choice since on my test env I was receiving a string type value, so I decided to go with `interface{}`

**Comment by mxpv** [comment]
ping @bmoffatt

**Comment by bmoffatt** [comment]
> @bmoffatt I'm not quite sure whether `map[string]string` is the right choice since on my test env I was receiving a string type value, so I decided to go with `interface{}`

👍

**Pull request merged**